### PR TITLE
Fix #3971: [RTL] Add style for all the TextViews.

### LINKED ISSUE #3971: Ensure all TextViews have style attribute set.
Use the style/theme for all the TextViews, to centrally manage the text alignment for RTL/LTR where the texts are aligned to extreme right or left and the texts that are center-aligned.

### PR DESCRIPTION
**Pull request by veena14cs**
## Explanation
This PR fixes #3971 . This PR  ensures that all the Textviews have style attribute set, to generally manage the text alignment for RTL/LTR where the texts are aligned to  right or left and the texts that are center-aligned.

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

File: app/src/main/res/layout-land/topic_practice_header_view.xml [hunk 1]
```diff
@@ -19,24 +19,23 @@
 
     <TextView
       android:id="@+id/master_skills_text_view"
+      style="@style/Heading2"
       android:layout_width="wrap_content"
       android:layout_height="wrap_content"
```
Comment (rt4914) [inline app/src/main/res/layout-land/topic_practice_header_view.xml]: layout_width and layout_height can be removed from this and other files as it is `wrap_content`

File: app/src/main/res/layout-land/topic_practice_header_view.xml [hunk 2]
```diff
@@ -19,24 +19,23 @@
...
-      android:textSize="20sp"
+      android:textAlignment="viewStart"
       app:layout_constraintStart_toStartOf="parent"
       app:layout_constraintTop_toTopOf="parent" />
 
     <TextView
       android:id="@+id/skills_description_text_view"
+      style="@style/Subtitle1"
       android:layout_width="wrap_content"
       android:layout_height="wrap_content"
```
Comment (rt4914) [inline app/src/main/res/layout-land/topic_practice_header_view.xml]: layout_width and layout_height can be removed from this and other files as it is `wrap_content`

File: app/src/main/res/layout-sw600dp-land/topic_lessons_title.xml [hunk 3]
```diff
@@ -17,12 +17,10 @@
 
     <TextView
       android:id="@+id/topic_play_text_view"
+      style="@style/Heading2"
       android:layout_width="528dp"
       android:layout_height="wrap_content"
```
Comment (rt4914) [inline app/src/main/res/layout-sw600dp-land/topic_lessons_title.xml]: layout_width and layout_height can be removed from this and other files as it is `wrap_content`

File: app/src/main/res/layout-sw600dp-land/topic_practice_header_view.xml [hunk 4]
```diff
@@ -22,24 +22,23 @@
 
       <TextView
         android:id="@+id/master_skills_text_view"
+        style="@style/Heading2"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
```
Comment (rt4914) [inline app/src/main/res/layout-sw600dp-land/topic_practice_header_view.xml]: layout_width and layout_height can be removed from this and other files as it is `wrap_content`

File: app/src/main/res/layout-sw600dp-land/topic_practice_header_view.xml [hunk 5]
```diff
@@ -22,24 +22,23 @@
...
-        android:textSize="20sp"
+        android:textAlignment="viewStart"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toTopOf="parent" />
 
       <TextView
         android:id="@+id/skills_description_text_view"
+        style="@style/Subtitle1"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
```
Comment (rt4914) [inline app/src/main/res/layout-sw600dp-land/topic_practice_header_view.xml]: layout_width and layout_height can be removed from this and other files as it is `wrap_content`

File: app/src/main/res/layout-sw600dp-port/topic_lessons_title.xml [hunk 6]
```diff
@@ -17,12 +17,10 @@
 
     <TextView
       android:id="@+id/topic_play_text_view"
+      style="@style/Heading2"
       android:layout_width="528dp"
       android:layout_height="wrap_content"
```
Comment (rt4914) [inline app/src/main/res/layout-sw600dp-port/topic_lessons_title.xml]: layout_height can be removed from this and other files as it is `wrap_content`

File: app/src/main/res/layout-sw600dp-port/topic_practice_header_view.xml [hunk 7]
```diff
@@ -22,24 +22,23 @@
 
       <TextView
         android:id="@+id/master_skills_text_view"
+        style="@style/Heading2"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
```
Comment (rt4914) [inline app/src/main/res/layout-sw600dp-port/topic_practice_header_view.xml]: layout_width and layout_height can be removed from this and other files as it is `wrap_content`

File: app/src/main/res/layout/profile_reset_pin_activity.xml [hunk 8]
```diff
@@ -62,14 +62,13 @@
 
           <TextView
             android:id="@+id/profile_reset_pin_main"
+            style="@style/Body"
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
```
Comment (rt4914) [inline app/src/main/res/layout/profile_reset_pin_activity.xml:67]: layout_width and layout_height can be removed from this and other files as it is `wrap_content`


Ditto for all other files.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@veena14cs PTAL

File: app/src/main/res/layout-land/topic_practice_header_view.xml (hunk 1 above)
Comment (veena14cs) [reply app/src/main/res/layout-land/topic_practice_header_view.xml]: Done.

File: app/src/main/res/layout-land/topic_practice_header_view.xml (hunk 2 above)
Comment (veena14cs) [reply app/src/main/res/layout-land/topic_practice_header_view.xml]: Done.

File: app/src/main/res/layout-sw600dp-land/topic_lessons_title.xml (hunk 3 above)
Comment (veena14cs) [reply app/src/main/res/layout-sw600dp-land/topic_lessons_title.xml]: Done.

File: app/src/main/res/layout-sw600dp-land/topic_practice_header_view.xml (hunk 4 above)
Comment (veena14cs) [reply app/src/main/res/layout-sw600dp-land/topic_practice_header_view.xml]: Done.

File: app/src/main/res/layout-sw600dp-land/topic_practice_header_view.xml (hunk 5 above)
Comment (veena14cs) [reply app/src/main/res/layout-sw600dp-land/topic_practice_header_view.xml]: Done.

File: app/src/main/res/layout-sw600dp-port/topic_lessons_title.xml (hunk 6 above)
Comment (veena14cs) [reply app/src/main/res/layout-sw600dp-port/topic_lessons_title.xml]: Done.

File: app/src/main/res/layout-sw600dp-port/topic_practice_header_view.xml (hunk 7 above)
Comment (veena14cs) [reply app/src/main/res/layout-sw600dp-port/topic_practice_header_view.xml]: Done.

File: app/src/main/res/layout/profile_reset_pin_activity.xml (hunk 8 above)
Comment (veena14cs) [reply app/src/main/res/layout/profile_reset_pin_activity.xml:67]: Done.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

File: app/src/main/res/layout-land/topic_practice_header_view.xml [hunk 9]
```diff
@@ -19,24 +19,19 @@
...
       android:id="@+id/skills_description_text_view"
-      android:layout_width="wrap_content"
-      android:layout_height="wrap_content"
+      style="@style/Subtitle1"
       android:layout_marginTop="4dp"
-      android:fontFamily="sans-serif"
       android:text="@string/topic_practice_skills_description"
       android:textColor="@color/oppiaSecondaryText"
-      android:textSize="16sp"
+      android:textAlignment="viewStart"
```
Comment (BenHenning) [inline app/src/main/res/layout-land/topic_practice_header_view.xml]: One of the goals with having central styles is to ensure that all RTL properties are defined through the styles rather than on the text view. Is this something that can be correctly moved to Subtitle1 (per other uses of it) or should this text view use a different style?

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @veena14cs. This & other PRs actually go into a lot more detail than I was expecting--all I think we actually need to do is ensure RTL-specific properties are defined in styles. The other properties can be moved later once we have a chance to figure out exactly what styles should exist at the top level.

FWIW I'm fine with keeping the current code (other than my comments about text alignment & other RTL properties), but I wanted to clarify expectations in case they weren't clear.

Same feedback for your other PRs. Please assign all back once addressed.

File: app/src/main/res/layout-land/topic_practice_header_view.xml (hunk 9 above)
Comment (veena14cs) [reply app/src/main/res/layout-land/topic_practice_header_view.xml]: To support LTR/RTL and center alignment I have changed the same versions of style with textalignment and gravity attribute. Please check if this makes sense. Once these file changes are approved I will make changes to other files in other PRs.
Note: All the styles for example `Heading1` will be changed to `HeadingCenter1` and `HeadingViewStart1` so final will have only 2 versions of Heading1.

Please let me know if we can propose this solution regardless of other attributes the focus is on only center alignment and right/left alignments .

File: app/src/main/res/layout-land/topic_lessons_title.xml [hunk 10]
```diff
@@ -16,14 +16,10 @@
 
     <TextView
       android:id="@+id/topic_play_text_view"
-      android:layout_width="wrap_content"
-      android:layout_height="wrap_content"
+      style="@style/Heading2"
```
Comment (BenHenning) [inline app/src/main/res/layout-land/topic_lessons_title.xml]: Doesn't this need to be a view start or center variant? Ditto elsewhere.

(In general, I'd expect all text views to have to pick some view start or view center variant to be explicit about their style).

File: app/src/main/res/values/styles.xml [hunk 11]
```diff
@@ -202,10 +202,25 @@
     <item name="fontFamily">sans-serif-medium</item>
   </style>
 
-  <style name="Heading3" parent="TextView.Common1">
+  <style name="HeadingViewStart2" parent="TextView.Common1">
```
Comment (BenHenning) [inline app/src/main/res/values/styles.xml]: Suggest inheriting from Heading2 and just defining the new property. Ditto for the other new properties.

File: app/src/main/res/values/styles.xml [hunk 12]
```diff
@@ -202,10 +202,25 @@
...
+    <item name="android:fontFamily">sans-serif-medium</item>
+    <item name="android:textSize">20sp</item>
+    <item name="fontFamily">sans-serif-medium</item>
+    <item name="android:textAlignment">viewStart</item>
+  </style>
+
+  <style name="HeadingCenter3" parent="TextView.Common1">
```
Comment (BenHenning) [inline app/src/main/res/values/styles.xml]: Suggest HeadingViewCenter3 to have parity with the view start variant. Ditto for the others.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @veena14cs. Left some high-level thoughts, but I think this is definitely on the right track.

My other high-level comment is that we should also be adding generic styles like TextViewCenter and TextViewStart that just define the center/view start properties (to be used by TextViews which don't have styles set today to ensure that all TextViews are covered by centrally-managed RTL styles). While this will add a lot more boilerplate styles, we can resolve these as we clean up styles as part of other ongoing projects.

@rt4914 please let me know if you have any concerns with this.

File: app/src/main/res/values/styles.xml (hunk 12 above)
Comment (veena14cs) [reply app/src/main/res/values/styles.xml]: Done.

File: app/src/main/res/values/styles.xml (hunk 11 above)
Comment (veena14cs) [reply app/src/main/res/values/styles.xml]: Done,

File: app/src/main/res/layout-land/topic_lessons_title.xml (hunk 10 above)
Comment (veena14cs) [reply app/src/main/res/layout-land/topic_lessons_title.xml]: Done.

**Comment by veena14cs** [comment]
> Thanks @veena14cs. Left some high-level thoughts, but I think this is definitely on the right track.
> 
> My other high-level comment is that we should also be adding generic styles like TextViewCenter and TextViewStart that just define the center/view start properties (to be used by TextViews which don't have styles set today to ensure that all TextViews are covered by centrally-managed RTL styles). While this will add a lot more boilerplate styles, we can resolve these as we clean up styles as part of other ongoing projects.
> 
> @rt4914 please let me know if you have any concerns with this.

I have added missing styles and fixed styles that have alignment properties. PTAL.

**Review by rt4914 — COMMENTED** [review]
@BenHenning Let me know your thoughts on this.

`Heading2ViewStart` should be named as `Heading2.ViewStart`
`Subtitle1ViewStart` should be named as `Subtitle1.ViewStart`

Same for all other styles.

File: app/src/main/res/layout-land/ongoing_topic_item.xml [hunk 13]
```diff
@@ -57,18 +57,17 @@
...
           android:id="@+id/topic_name_text_view"
+          style="@style/TextViewStart"
           android:layout_width="match_parent"
           android:layout_height="wrap_content"
           android:layout_marginStart="8dp"
           android:layout_marginTop="8dp"
           android:layout_marginEnd="8dp"
           android:ellipsize="end"
           android:fontFamily="sans-serif"
-          android:gravity="start"
```
Comment (BenHenning) [inline app/src/main/res/layout-land/ongoing_topic_item.xml:67]: TextViewStart only specifies textAlignment; should the gravity property stay or does it also need to be part of the style? Though, I am wondering whether this is even used given that the text view is in a ConstraintLayout.

File: app/src/main/res/layout/profile_chooser_profile_view.xml [hunk 14]
```diff
@@ -79,6 +80,7 @@
 
         <TextView
           android:id="@+id/profile_is_admin_text"
+          style="@style/TextViewCenterHorizontal"
```
Comment (BenHenning) [inline app/src/main/res/layout/profile_chooser_profile_view.xml]: Isn't this conditional? Ditto for profile_last_visited.

File: app/src/main/res/layout-sw600dp-port/profile_chooser_fragment.xml [hunk 15]
```diff
@@ -43,8 +43,7 @@
           app:layout_constraintTop_toTopOf="parent" />
 
         <TextView
-          style="@style/Subtitle1"
-          android:gravity="center"
+          style="@style/Subtitle1ViewCenterVertical"
```
Comment (BenHenning) [inline app/src/main/res/layout-sw600dp-port/profile_chooser_fragment.xml]: Given that android:gravity="center" is both horizontal & vertical, do we need a style that sets both for these cases?

File: app/src/main/res/layout-sw600dp-land/ongoing_topic_item.xml [hunk 16]
```diff
@@ -55,17 +55,16 @@
...
           android:id="@+id/topic_name_text_view"
+          style="@style/TextViewStart"
           android:layout_width="match_parent"
           android:layout_height="wrap_content"
           android:layout_marginStart="16dp"
           android:layout_marginTop="8dp"
           android:layout_marginEnd="16dp"
           android:ellipsize="end"
           android:fontFamily="sans-serif"
-          android:gravity="start"
```
Comment (BenHenning) [inline app/src/main/res/layout-sw600dp-land/ongoing_topic_item.xml:65]: Both gravity & text alignment is used here--which is correct?

File: app/src/main/res/layout-land/section_title.xml [hunk 17]
```diff
@@ -25,14 +25,10 @@
 
     <TextView
       android:id="@+id/section_title_text_view"
-      android:layout_width="wrap_content"
-      android:layout_height="wrap_content"
```
Comment (BenHenning) [inline app/src/main/res/layout-land/section_title.xml]: Why are these removed? They aren't explicitly defined in the style.

Ditto elsewhere.

File: app/src/main/res/layout-land/section_title.xml [hunk 18]
```diff
@@ -25,14 +25,10 @@
...
+      style="@style/Heading3ViewStart"
       android:layout_marginStart="72dp"
       android:layout_marginTop="28dp"
       android:layout_marginEnd="72dp"
-      android:fontFamily="sans-serif-medium"
-      android:text="@{viewModel.sectionTitleText}"
-      android:textColor="@color/oppiaPrimaryText"
```
Comment (BenHenning) [inline app/src/main/res/layout-land/section_title.xml]: This also doesn't seem to be defined in the style.

File: app/src/main/res/layout/profile_reset_pin_activity.xml [hunk 19]
```diff
@@ -62,14 +62,11 @@
 
           <TextView
             android:id="@+id/profile_reset_pin_main"
-            android:layout_width="wrap_content"
-            android:layout_height="wrap_content"
```
Comment (BenHenning) [inline app/src/main/res/layout/profile_reset_pin_activity.xml]: Body uses different layout width/height properties than this--is it the correct style to be using? This TextView also didn't define an explicit font family before.

File: app/src/main/res/values/styles.xml [hunk 20]
```diff
@@ -339,4 +413,21 @@
     <item name="fontFamily">sans-serif</item>
     <item name="textAllCaps">false</item>
   </style>
+
+  <style name="TextViewStart" parent="TextView.Common1">
+    <item name="android:textAlignment">viewStart</item>
+  </style>
+
+  <style name="TextViewCenter" parent="TextView.Common1">
```
Comment (BenHenning) [inline app/src/main/res/values/styles.xml:450]: Would 'CenterVerticalHorizontal' be clearer to distinguish from the other two center options? Ditto for other similar styles.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @veena14cs! This looks much closer to what I was thinking. Just had a few follow-up comments--PTAL.

**Comment by BenHenning** [comment]
> @BenHenning Let me know your thoughts on this.
> 
> `Heading2ViewStart` should be named as `Heading2.ViewStart` `Subtitle1ViewStart` should be named as `Subtitle1.ViewStart`
> 
> Same for all other styles.

@rt4914 sure; I don't have a strong stance here since the styles are temporary (though given that they may be around for a bit while we figure out the final way to define styles, it seems reasonable).

@veena14cs please also update the PR to address the latest conflicts.

File: app/src/main/res/layout-land/topic_practice_header_view.xml (hunk 9 above)
Comment (BenHenning) [reply app/src/main/res/layout-land/topic_practice_header_view.xml]: Per the latest changes, the solution seems good.

File: app/src/main/res/layout-land/section_title.xml (hunk 17 above)
Comment (veena14cs) [reply app/src/main/res/layout-land/section_title.xml]: Reverted.

File: app/src/main/res/layout-land/section_title.xml (hunk 18 above)
Comment (veena14cs) [reply app/src/main/res/layout-land/section_title.xml]: Reverted

File: app/src/main/res/layout/profile_reset_pin_activity.xml (hunk 19 above)
Comment (veena14cs) [reply app/src/main/res/layout/profile_reset_pin_activity.xml]: Fixed style for this view.

File: app/src/main/res/layout/profile_chooser_profile_view.xml (hunk 14 above)
Comment (veena14cs) [reply app/src/main/res/layout/profile_chooser_profile_view.xml]: How to set conditional style for the views? I found that is is not doable https://code.google.com/archive/p/android-developer-preview/issues/2613.

File: app/src/main/res/layout-sw600dp-port/profile_chooser_fragment.xml (hunk 15 above)
Comment (veena14cs) [reply app/src/main/res/layout-sw600dp-port/profile_chooser_fragment.xml]: Yes added `Subtitle1Center` style.

File: app/src/main/res/layout-land/ongoing_topic_item.xml (hunk 13 above)
Comment (veena14cs) [reply app/src/main/res/layout-land/ongoing_topic_item.xml:67]: It is not being used so removed it.

File: app/src/main/res/layout-sw600dp-land/ongoing_topic_item.xml (hunk 16 above)
Comment (veena14cs) [reply app/src/main/res/layout-sw600dp-land/ongoing_topic_item.xml:65]: android:gravity="start" doesn't work for RTL. This line doesn't seem to make any difference it is unused.

File: app/src/main/res/values/styles.xml (hunk 20 above)
Comment (veena14cs) [reply app/src/main/res/values/styles.xml:450]: Should I rename it to `TextViewCenterVerticalHorizontal` or just `CenterVerticalHorizontal`?

Comment (veena14cs) [reply app/src/main/res/values/styles.xml:450]: Also, I think `TextViewCenter` seems like more general as it defines by default both the vertical and horizontal axis.

**Comment by veena14cs** [comment]
> > @BenHenning Let me know your thoughts on this.
> > `Heading2ViewStart` should be named as `Heading2.ViewStart` `Subtitle1ViewStart` should be named as `Subtitle1.ViewStart`
> > Same for all other styles.
> 
> @rt4914 sure; I don't have a strong stance here since the styles are temporary (though given that they may be around for a bit while we figure out the final way to define styles, it seems reasonable).
> 
> @veena14cs please also update the PR to address the latest conflicts.

@rt4914 I agree with @BenHenning as this is just a temporary solution and I would keep it as it is for now.

**Review by rt4914 — APPROVED** [review]
Sounds good. Lets keep it as it is for now. Thanks.

**Comment by veena14cs** [comment]
> Sounds good. Lets keep it as it is for now. Thanks.

Thanks @rt4914 .

File: app/src/main/res/layout/profile_chooser_profile_view.xml (hunk 14 above)
Comment (veena14cs) [reply app/src/main/res/layout/profile_chooser_profile_view.xml]: I created new style called `SubtitleLight` that matches the Textviews property without having gravity attribute.

File: app/src/main/res/layout-sw600dp-land/profile_chooser_add_view.xml [hunk 21]
```diff
@@ -55,6 +55,7 @@
 
         <TextView
           android:id="@+id/add_profile_text"
+          style="@style/TextViewCenter"
```
Comment (BenHenning) [inline app/src/main/res/layout-sw600dp-land/profile_chooser_add_view.xml]: This isn't being centered in the layout--is this correct?

File: app/src/main/res/layout/profile_progress_header.xml [hunk 22]
```diff
@@ -90,6 +90,7 @@
 
         <TextView
           android:id="@+id/ongoing_topics_count"
+          style="@style/TextViewCenter"
```
Comment (BenHenning) [inline app/src/main/res/layout/profile_progress_header.xml:93]: Ditto here--this didn't have a center property before.

File: app/src/main/res/layout-land/profile_chooser_profile_view.xml [hunk 23]
```diff
@@ -61,26 +61,20 @@
 
         <TextView
           android:id="@+id/profile_last_visited"
+          style="@style/SubtitleLight1"
```
Comment (BenHenning) [inline app/src/main/res/layout-land/profile_chooser_profile_view.xml]: Shouldn't this be a 'Center'? Ditto for other profile_chooser_profile_view layouts, and for profile_is_admin_text below.

File: app/src/main/res/layout/profile_chooser_fragment.xml [hunk 24]
```diff
@@ -100,7 +99,7 @@
       app:layout_constraintEnd_toEndOf="parent">
 
       <TextView
-        style="@style/Subtitle1"
+        style="@style/Subtitle1ViewCenter"
         android:layout_marginBottom="24dp"
         android:gravity="center"
```
Comment (BenHenning) [inline app/src/main/res/layout/profile_chooser_fragment.xml]: Can this be removed per the latest style?

File: app/src/main/res/layout/profile_reset_pin_activity.xml (hunk 8 above)
Comment (BenHenning) [reply app/src/main/res/layout/profile_reset_pin_activity.xml:67]: It seems this was wrong per the latest changes (& was reverted). @rt4914 can you confirm whether this comment is still relevant per the latest changes?

File: app/src/main/res/layout/profile_chooser_profile_view.xml (hunk 14 above)
Comment (BenHenning) [reply app/src/main/res/layout/profile_chooser_profile_view.xml]: The style thing is a bit of a conundrum. I think defaulting the property is probably the best that we can do, but it's probably going to end up causing some confusion when we figure out the long-term styles & add the CI check. Can you add a comment here & to other versions of this text view (ie in other versions of this file) to explain why this style is used vs. one that defines the text alignment?

File: app/src/main/res/values/styles.xml (hunk 20 above)
Comment (BenHenning) [reply app/src/main/res/values/styles.xml:450]: Hmm after taking another pass, I think this is fine for the medium term.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @veena14cs. Took a pass on the latest changes--PTAL.

File: app/src/main/res/layout/profile_progress_header.xml (hunk 22 above)
Comment (veena14cs) [reply app/src/main/res/layout/profile_progress_header.xml:93]: To keep it consistent I added center property here. And by default it was center aligned based on the layout. Same with other files.

File: app/src/main/res/layout-land/profile_chooser_profile_view.xml (hunk 23 above)
Comment (veena14cs) [reply app/src/main/res/layout-land/profile_chooser_profile_view.xml]: No, as the gravity is handled conditionally.

File: app/src/main/res/layout/profile_chooser_fragment.xml (hunk 24 above)
Comment (veena14cs) [reply app/src/main/res/layout/profile_chooser_fragment.xml]: Done.

File: app/src/main/res/layout/profile_chooser_profile_view.xml (hunk 14 above)
Comment (veena14cs) [reply app/src/main/res/layout/profile_chooser_profile_view.xml]: Added comment please check if that sounds good.

File: app/src/main/res/layout-sw600dp-land/profile_chooser_add_view.xml (hunk 21 above)
Comment (veena14cs) [reply app/src/main/res/layout-sw600dp-land/profile_chooser_add_view.xml]: No, fixed it.

File: app/src/main/res/values/styles.xml [hunk 25]
```diff
@@ -348,7 +352,7 @@
     <item name="android:textAlignment">viewStart</item>
   </style>
 
-  <style name="CaptionViewCenter" parent="Caption">
+  <style name="CaptionViewCenterVertical" parent="Caption">
```
Comment (BenHenning) [inline app/src/main/res/values/styles.xml]: No views were updated when this was changed--is this actually being used anywhere?

File: app/src/main/res/layout-land/profile_chooser_profile_view.xml [hunk 26]
```diff
@@ -61,26 +61,20 @@
...
-          android:gravity="center"
           android:textColor="@color/white"
-          android:textSize="12sp"
-          android:textStyle="italic"
           android:visibility="@{viewModel.profile.lastLoggedInTimestampMs > 0 ? View.VISIBLE : View.GONE}"
           profile:lastVisited="@{viewModel.profile.lastLoggedInTimestampMs}" />
 
         <TextView
           android:id="@+id/profile_is_admin_text"
+          style="@style/SubtitleLight1"
```
Comment (BenHenning) [inline app/src/main/res/layout-land/profile_chooser_profile_view.xml]: Should this be the centered version since the gravity was center before?

File: app/src/main/res/layout/profile_chooser_add_view.xml [hunk 27]
```diff
@@ -59,6 +60,7 @@
 
         <TextView
           android:id="@+id/add_profile_description_text"
+          style="@style/TextViewStart"
```
Comment (BenHenning) [inline app/src/main/res/layout/profile_chooser_add_view.xml:63]: Should this be centered per the layout_gravity? We did this for an earlier layout in the PR.

File: app/src/main/res/layout/profile_chooser_profile_view.xml (hunk 14 above)
Comment (BenHenning) [reply app/src/main/res/layout/profile_chooser_profile_view.xml]: It looks good--thanks!

File: app/src/main/res/layout/profile_progress_header.xml (hunk 22 above)
Comment (BenHenning) [reply app/src/main/res/layout/profile_progress_header.xml:93]: Ah I see--that wasn't very clear. Thanks for clarifying.

**Review by BenHenning — COMMENTED** [review]
Thanks @veena14cs. Just a few comments left. Some of them might be correct as-is per your earlier comments, but I just wanted to double check.

File: app/src/main/res/values/styles.xml (hunk 25 above)
Comment (veena14cs) [reply app/src/main/res/values/styles.xml]: I have updated changes some Textviews were missed.

File: app/src/main/res/layout-land/profile_chooser_profile_view.xml (hunk 26 above)
Comment (veena14cs) [reply app/src/main/res/layout-land/profile_chooser_profile_view.xml]: Fixed.

File: app/src/main/res/layout/profile_chooser_add_view.xml (hunk 27 above)
Comment (veena14cs) [reply app/src/main/res/layout/profile_chooser_add_view.xml:63]: `layout_gravity` seems unused here as gravity is controlled by parent layout, so removed it. The description is start aligned as per the UI.

**Comment by veena14cs** [comment]
> Thanks @veena14cs. Just a few comments left. Some of them might be correct as-is per your earlier comments, but I just wanted to double check.

@BenHenning I have made changes and addressed your comments PTAL.

File: app/src/main/res/layout/profile_reset_pin_activity.xml (hunk 8 above)
Comment (rt4914) [reply app/src/main/res/layout/profile_reset_pin_activity.xml:67]: This comment can be closed now. Thanks.

**Comment by rt4914** [comment]
@BenHenning and @veena14cs I have answered/resolved to all the comment threads that were started by me. Thanks.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Review by BenHenning — APPROVED** [review]
Thanks @veena14cs! I have nothing else to add. I think this is a really nice change toward centrally managing RTL properties, and sets us up for more cleanly organizing centralized styles & enforcing their presence in future work.

**Comment by BenHenning** [comment]
> @BenHenning and @veena14cs I have answered/resolved to all the comment threads that were started by me. Thanks.

Thanks @rt4914!

**Comment by BenHenning** [comment]
Merging since there are no more open conversation threads & everything is approved.
